### PR TITLE
Add support for decoding tiff's with float pixel data

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This **PR** adds support for decoding Tiff images with float pixel data.

- Add support for decoding tiff's with 32bit float rgb pixel data
- Add support for decoding tiff's with 32bit float gray pixel data with min is black
- Add support for decoding tiff's with 32bit float gray pixel data with min is white

### CODE REVIEW

File: src/ImageSharp/Formats/Tiff/PhotometricInterpretation/BlackIsZero32FloatTiffColor{TPixel}.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            int offset = 0;
+            for (int y = top; y < top + height; y++)
+            {
+                Span<TPixel> pixelRow = pixels.GetRowSpan(y).Slice(left, width);
+                if (this.isBigEndian)
+                {
+                    for (int x = 0; x < pixelRow.Length; x++)
+                    {
+                        data.Slice(offset, 4).CopyTo(buffer);
+                        Array.Reverse(buffer);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Tiff/PhotometricInterpretation/BlackIsZero32FloatTiffColor{TPixel}.cs:44]: Can we not use `BinaryPrimitives.ReadSingleBigEndian` here and elsewhere?

Comment (brianpopow) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/BlackIsZero32FloatTiffColor{TPixel}.cs:44]: `ReadSingleBigEndian` is only available with `net5.0` and above

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Tiff/PhotometricInterpretation/BlackIsZero32FloatTiffColor{TPixel}.cs:44]: Didn't know that. Looking at the source we could only polyfill back to .NET Core 2.0 also. We'll leave it like this for now.

**Review by JimBobSquarePants — APPROVED** [review]
Great stuff 👍

**Pull request merged**